alternator/ttl.cc

```diff
@@ -568,7 +569,17 @@ static future<> scan_table_ranges(
                     std::current_exception());
             }
             // If we didn't break out of this loop, add a minimal sleep
-            co_await seastar::sleep(1s);
+            if (retries >= 10) {
+                // Don't get stuck forever asking the same page, maybe there's
+                // a bug or a real problem in several replicas. Give up on
+                // this scan an retry the scan from a random position later,
+                // in the next scan period.
+                throw runtime_exception("scanner thread failed after too many timeouts for the same page");
+            }
+            co_await sleep_abortable(std::chrono::seconds(1), abort_source);
+            if (abort_source.abort_requested()) {
```

Comment:
co_await sleep_abortable will throw if timed out so the if statement won't be executed...

See https://github.com/scylladb/seastar/blob/96e566bdfee7fff493c7e25c869a2c252aabfac7/include/seastar/core/sleep.hh#L80